### PR TITLE
EN-12140 Also write X-SODA2-Fields in csv export.

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala [hunk 1]
```diff
@@ -20,7 +23,33 @@ trait Exporter {
   protected def exporterHeaders(schema: ExportDAO.CSchema): HttpResponse =
     schema.lastModified.fold(NoOp) { lm =>
       Header("Last-Modified", HttpUtils.HttpDateFormat.print(lm))
+      maybeSoda2FieldsHeader(schema)
     }
+
+  /**
+    * Override in specific exporters if we want X-SODA2-Fields
+    */
```
Comment (aerust) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala]: I think I'm just missing it here, but what does the caller have to do to get the X-SODA2-Fields?

**Comment by aerust** [comment]
I think added a bit of an explanation to the commit message of _why_ you are making this change would be good.

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala (hunk 1 above)
Comment (chitang) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala]: Override mayBeSoda2FieldsHeader to call writeSoda2FieldsHeader.  That is done in JsonExporter and TableExporter.

Comment (aerust) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala]: But will it be always that they are added?

Comment (chitang) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala]: There are other unique exporters (cjson, soqlpack...) in nbe that I still don't want to give in.

Comment (aerust) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/export/Exporter.scala]: Oh, okay. :+1:

**Comment by aerust** [comment]
Commit message is nice :) looks good to me :sheep:

**Pull request merged**